marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/CallBatcherImpl.java

```diff
@@ -457,7 +465,16 @@ private void initialize() {
             logger.warn("threadCount should be 1 or greater -- setting threadCount to number of hosts: " + threadCount);
         }
         if (initialized.getAndSet(true)) return;
-// TODO
+        
+        if(inputType == Void.class) {
```

Comment:
In looking at the code, I'm wondering belatedly whether the operation might be easier to understand if the test was on

`callArgsGenerator != null`

instead of on 

    `inputType == Void.class` 

both here and also on line 708 (but keeping the Void test on line 296).

What do you think?

Reply:
Yes. we can do that.